src/Commands/Domain/SetPrimaryCommand.php

```diff
@@ -0,0 +1,104 @@
+use Pantheon\Terminus\Models\Environment;
+use Pantheon\Terminus\Models\Site;
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class SetPrimaryCommand
+ * @package Pantheon\Terminus\Commands\Domain
+ */
+class SetPrimaryCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+    use WorkflowProcessingTrait;
+
+    /**
+     * Sets the primary domain for a site and environment, causing all traffic to redirect to the primary domain.
+     *
+     * @authorize
+     *
+     * @command domain:primary:set
+     *
+     * @param string $site_env Site & environment in the format `site-name.env`
+     * @param string $domain A domain that has been associated to your site. Optional when running interactively.
+     *
+     * @usage domain:primary:set <site_env>
+     */
+    public function set($site_env, $domain)
+    {
+        /**
+         * @var $site Site
+         * @var $env Environment
+         */
+        list(, $env) = $this->getSiteEnv($site_env);
+
+        // The primary domain is set via a workflow so as to use workflow logging to track changes.
+        $this->log()->notice('Setting primary domain to {domain}...', ['domain' => $domain]);
+        $workflow = $env->setPrimaryDomain($domain);
+        $this->processWorkflow($workflow);
+    }
+
+    /**
+     * Removes the primary domain for a site and environment.
+     *
+     * @authorize
+     *
+     * @command domain:primary:reset
+     *
+     * @param string $site_env Site & environment in the format `site-name.env`
+     */
+    public function reset($site_env)
+    {
+        /**
+         * @var $site Site
+         * @var $env Environment
+         */
+        list(, $env) = $this->getSiteEnv($site_env);
+
+        $this->log()->notice('Unsetting primary domain...');
+        $workflow = $env->setPrimaryDomain(null);
+        $this->processWorkflow($workflow);
+    }
+
+    /**
+     * Prompt the user for the domain, if it was not specified.
+     *
+     * n.b. This hook is not called in --no-interaction mode.
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @param AnnotationData $annotationData
+     *
+     * @hook interact domain:primary:set
+     */
+    public function setInteract(InputInterface $input, OutputInterface $output, AnnotationData $annotationData)
+    {
+        $domain = $input->getArgument('domain');
+        if (empty($domain)) {
+            /**
+             * @var $site Site
+             * @var $env Environment
+             */
+            list($site, $env) = $this->getSiteEnv($input->getArgument('site_env'));
+            $domains = array_filter($env->getDomains()->ids(), function ($domain) {
+                return !preg_match('|\.pantheonsite\.io$|', $domain);
```

Comment:
Is there a reason why we don't want to allow platform domains to be used as primary? I could imagine a situation where it can be useful during testing.

Reply:
Greg & I had extensive back and forth about this in this PR's conversation, the git log associated with the current state of things has the culmination of that. Maybe we could add it back into the list if the environment isn't live.

Reply:
We'll want to extract this domain and move it somewhere, probably into the consts, so that it can be located and changed if necessary.